### PR TITLE
chore: release notes 0.38.0

### PR DESCRIPTION
**Pull request by azhou-determined**
## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/release-notes.rst [hunk 1]
```diff
@@ -6,6 +6,364 @@
...
+   Amazon RDS for PostgreSQL. Changes to the deployment code will ensure this transition to the new
+   default.
+
+-  Database: As a follow-up to the earlier notice, PostgreSQL 12 will reach End of Life on November
+   14, 2024. Instances still using PostgreSQL 12 or earlier should upgrade to PostgreSQL 13 or later
+   to maintain compatibility. The application will log a warning if it detects a connection to any
+   PostgreSQL version older than 12, and this warning will be updated to include PostgreSQL 12 once
+   it is End of Life.
+
+**************
```
Comment (mackrorysd) [inline docs/release-notes.rst:188]: Looks like we've got some duplication? Not referring to the heading - I see that was duplicated the same way previously - but the whole list.

File: docs/release-notes.rst [hunk 2]
```diff
@@ -6,6 +6,364 @@
...
+-  Web UI: previously, trials that reported an ``epoch`` metric enabled an epoch X-axis in the Web
+   UI metrics tab. This metric name has been changed to ``epochs``, with ``epoch`` as a fallback
+   option.
+
+-  Cluster: A reminder that Amazon Aurora V1 will reach End of Life at the end of 2024. It is no
+   longer supported as the default persistent storage for AWS Determined deployments. We recommend
+   that users migrate to Amazon RDS for PostgreSQL. For more information, visit the `migration
+   instructions <https://gist.github.com/maxrussell/c67f4f7d586d55c4eb2658cc2dd1c290>`_.
+
+-  Cluster: After Amazon Aurora V1 reaches End of Life, support for Amazon Aurora V1 in ``det deploy
```
Comment (mackrorysd) [inline docs/release-notes.rst]: Feels like this item and the item above could be combined? Or at least they should both have the `database:` prefix, not Cluster?

File: docs/release-notes.rst (hunk 1 above)
Comment (azhou-determined) [reply docs/release-notes.rst]: yeah, i killed and reran rph. fixed now.

File: docs/release-notes.rst [hunk 3]
```diff
@@ -6,6 +6,364 @@
...
+
+-  Core API: the ``SearcherContext`` (``core.searcher``) has been deprecated. Training code no
+   longer requires ``core.searcher.operations`` to run, and progress should be reported through
+   ``core.train.report_progress``.
+
+-  DeepSpeed: the ``num_micro_batches_per_slot`` and ``train_micro_batch_size_per_gpu`` attributes
+   on ``DeepSpeedContext`` have been replaced with ``get_train_micro_batch_size_per_gpu()`` and
+   ``get_num_micro_batches_per_slot()``.
+
+-  Horovod: the horovod distributed training backend has been deprecated. Users are encouraged to
```
Comment (mackrorysd) [inline docs/release-notes.rst]: Capitalize Horovod in both instances?

File: docs/release-notes.rst (hunk 2 above)
Comment (azhou-determined) [reply docs/release-notes.rst]: nice catch, done

File: docs/release-notes.rst [hunk 4]
```diff
@@ -6,6 +6,181 @@
...
+      -  ``det token describe``: Show details of specific access tokens.
+      -  ``det token revoke``: Revoke an access token.
+
+   -  API:
+
+      -  ``POST /api/v1/tokens``: Create a new access token.
+      -  ``GET /api/v1/tokens``: Retrieve a list of access tokens.
+      -  ``PATCH /api/v1/tokens/{token_id}``: Edit an existing access token.
+
+-  API: introduce ``keras.DeterminedCallback``, a new high-level training API for TF Keras that
```
Comment (jgongd) [inline docs/release-notes.rst:61]: style: API: Introduce

**Review by mackrorysd — APPROVED** [review]

File: docs/release-notes.rst [hunk 5]
```diff
@@ -6,6 +6,181 @@
...
+   integrates Keras training code with Determined through a single :ref:`Keras Callback
+   <api-keras-ug>`.
+
+-  API: introduce ``deepspeed.Trainer``, a new high-level training API for DeepSpeedTrial that
```
Comment (jgongd) [inline docs/release-notes.rst:65]: Ditto

File: docs/release-notes.rst (hunk 4 above)
Comment (azhou-determined) [reply docs/release-notes.rst]: done for all of them
